compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -169,127 +172,239 @@ printSummary color res =
         ]
     printScenarioResults color res
 
-data TemplateIdentifier = TemplateIdentifier
+data ContractIdentifier = ContractIdentifier
     { package :: Maybe T.Text -- `package == Nothing` means local package
-    , qualifiedTemplate :: T.Text
+    , qualifiedName :: T.Text
     }
     deriving (Eq, Ord, Show)
 
 data ChoiceIdentifier = ChoiceIdentifier
-    { packageTemplate :: TemplateIdentifier
+    { packageContract :: ContractIdentifier
     , choice :: T.Text
     }
     deriving (Eq, Ord, Show)
 
+data Variety tpl iface = TemplateV tpl | InterfaceV iface
+    deriving (Eq, Ord, Show)
+
 data Report = Report
```

Comment:
it seems you're no longer constructing a value of this type, which makes me worry that the extraction logic and the display logic could get intertwined again

Reply:
Removed Variety (it was barely used), and Report